packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -399,7 +409,26 @@ class _HelperErrorState extends State<_HelperError> with SingleTickerProviderSta
   }
 
   Widget _buildError() {
-    assert(widget.error != null || widget.errorText != null);
+    assert(
+      _hasError,
+      'errorText or error must be provided'
+    );
```

Comment:
Thanks for the update!

Attaching asserts to the constructor is nice, but so is utilizing existing getters 👍